### ISSUE TITLE
ItemPageCcLicenseFieldComponent wrongly assumes that any license in `dc.rights` is a CC license

### ISSUE THREAD

**Issue by abelgomez** [DSpace/dspace-angular#3170]
## Describe the bug
The `ItemPageCcLicenseFieldComponent`, which is rendered for any untyped item, wrongly assumes that any value in `dc.rights` is a valid CC license.

## To Reproduce
Create an untyped item, set `dc.rights` and `dc.rights.uri`, and see a section in the item with the title "Creative Commons license".

![imagen](https://github.com/DSpace/dspace-angular/assets/5593904/244c94c7-bd25-42fa-bb87-3686f36c5ee6)
![imagen](https://github.com/DSpace/dspace-angular/assets/5593904/edaac59d-a035-49e3-999e-5f2306913e10)

## Expected behavior
The title of the license section should only mention Creative Commons if `dc.rights` is a valid CC license.

I think that `ItemPageCcLicenseFieldComponent` should be renamed to `ItemPageLicenseFieldComponent`. If `dc.rights.uri` points to a valid CC license, it should work as the current `ItemPageCcLicenseFieldComponent`. If `dc.rights`/`dc.rights.uri` have other type of license, the component should render the license text as specified in `dc.rights`.

## Related work

This should be considered before implementing #3167.

**Comment by amgciadev** [comment]
We have encountered this too and we solve it by simply amending the label: "item.page.cc.license.title": "Rights and licensing". Happy to submit a PR. Label item.page.cc.license.title could also be renamed to: item.page.license.title to be more generic?

**Comment by tdonohue** [comment]
@abelgomez : I'm not against a refactor here to try to determine if a license is actually a CC license before displaying it as such.  As long as it doesn't *break* existing sites which use `dc.rights` to store CC License (and I don't think it should), then it could be backported to 7.x and 8.x.

I'll assign this to you @abelgomez since it sounds like you want to claim it (to do in a few weeks time).

**Comment by abelgomez** [comment]
Hi @tdonohue , it won't break existing sites using the default template, but it *may* break existing sites with modifications or custom templates relying on the `ItemPageCcLicenseFieldComponent` since we don't know how developers have used it.

But I think it can be done with minimal impact, and in any case this kind of breaking changes are not rare (e.g., the migration to "standalone" components between Dspace 7 and 8 has been quite painful for me 😊).

We can address in this upcoming PR the related issue https://github.com/DSpace/dspace-angular/issues/3167.

**Comment by abelgomez** [comment]
Hi @tdonohue ,

I have started to work on this, and I'm near to have a proposal to fix this issue. Nevertheless, I'm still drafting the PR and the documentation.

In the meantime, I have a question: I started this work in the 8_x branch (since it was marked as in project [DSpace 8.x and 7.6.x Maintenance](https://github.com/orgs/DSpace/projects/29).

Should I create the PR against the main branch instead?

**Comment by tdonohue** [comment]
@abelgomez : Yes, I'd recommend PRs *almost always* be created against `main`.  The only exception is when the code changes do *not* affect the `main` branch.   So, please do update your PR to be against `main`.
